### PR TITLE
Fixed Hardcoded Dtype

### PR DESCRIPTION
**Pull request by qyy2003**
Solved #206 

It seems nearly all the hardcoded types are in test file. (Ignored right now.)

### CODE REVIEW

**Comment by qyy2003** [comment]
I have verified that the local CPU + remote disk setup works correctly with float16 on all four different serde methods.

The only remaining hardcoded element in the functional code is in the following snippet within the `validate_reverse_correctness` function, located in `lmcache-vllm/lmcache_vllm/utils/positional_encoding.py`:

```python
dumb_q = torch.rand((num_tokens, hidden_dim), device="cuda", dtype=torch.bfloat16)
dumb_k = torch.rand((num_tokens, hidden_dim), device="cuda", dtype=torch.bfloat16)
```

It seems to be test code, so I decided not to modify it.

**Review by YaoJiayi — COMMENTED** [review]
Thanks @qyy2003 ! But the unit tests are failing. Could you fix it first?
Another side note: could you compare the performance improvement from bfloat16 to datatypes ike fp8?

**Comment by qyy2003** [comment]
@YaoJiayi  Fixed unitest problem. Its system dtype is half but it rands float32 tensor.

**Comment by qyy2003** [comment]
Locally, the test is ok. I guess the reason it fails in github current is that it uses the original unitest code.

File: lmcache/storage_backend/serde/cachegen_decoder.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 @_lmcache_nvtx_annotate
 def quant(bins: int, xq: torch.Tensor, max1: float):
     C = bins // 2 - 1
-    x = xq / C * max1  # .to(torch.float16)
+    x = xq / C * max1  # .to(self.dtype)
```
Comment (ApostaC) [inline lmcache/storage_backend/serde/cachegen_decoder.py]: Maybe just remove the comment in this line?

**Review by ApostaC — APPROVED** [review]
LGTM, thanks! Just one minor comment, feel free to fix that if you want.

**Pull request merged**